gk/main.c

```diff
@@ -2301,3 +2327,77 @@ gk_flush_flow_table(const char *src_prefix,
 
 	return 0;
 }
+
+int
+gk_log_flow_state(const char *src_addr,
+	const char *dst_addr, struct gk_config *gk_conf)
+{
+	int ret;
+	struct ipaddr src;
+	struct ipaddr dst;
+	struct gk_log_request log;
+	struct mailbox *mb;
+	struct gk_cmd_entry *entry;
+
+	if (src_addr == NULL) {
+		GK_LOG(ERR, "Failed to log flow state: source address is NULL\n");
+		return -1;
+	}
+	if (dst_addr == NULL) {
+		GK_LOG(ERR, "Failed to log flow state: destination address is NULL\n");
+		return -1;
+	}
+
+	ret = convert_str_to_ip(src_addr, &src);
+	if (ret < 0) {
+		GK_LOG(ERR, "Failed to log flow state: source address (%s) is invalid\n",
+			src_addr);
+		return -1;
+	}
+
+	ret = convert_str_to_ip(dst_addr, &dst);
+	if (ret < 0) {
+		GK_LOG(ERR, "Failed to log flow state: destination address (%s) is invalid\n",
+			dst_addr);
+		return -1;
+	}
+
+	if (unlikely(src.proto != dst.proto)) {
+		GK_LOG(ERR, "Failed to log flow state: source (%s) and destination (%s) addresses don't have the same IP type\n",
+			src_addr, dst_addr);
+		return -1;
+	}
+
+	memset(&log, 0, sizeof(log));
+
+	strcpy(log.src_addr, src_addr);
+	strcpy(log.dst_addr, dst_addr);
+	log.flow.proto = src.proto;
+	if (log.flow.proto == RTE_ETHER_TYPE_IPV4) {
+		log.flow.f.v4.src = src.ip.v4;
+		log.flow.f.v4.dst = dst.ip.v4;
+	} else if (likely(log.flow.proto == RTE_ETHER_TYPE_IPV6)) {
+		log.flow.f.v6.src = src.ip.v6;
+		log.flow.f.v6.dst = dst.ip.v6;
+	} else
+		return -1;
+
+	mb = get_responsible_gk_mailbox(&log.flow, gk_conf);
+	if (mb == NULL)
+		return -1;
+
+	entry = mb_alloc_entry(mb);
+	if (entry == NULL) {
+		GK_LOG(WARNING,
+			"Cannot allocate an entry for the mailbox of the GK block to log flow state that match src_addr=%s and dst_addr=%s\n",
+			src_addr, dst_addr);
+		return -1;
+	}
+
+	entry->op = GK_LOG_FLOW_STATE;
+	entry->u.log = log;
```

Comment:
Instead of copying the source and destination addresses to `log` in the code before the lines above, just copy them here directly to `entry`.

Reply:
We need the ``flow`` to first get the responsible mailbox, then we can assign the ``flow`` to the ``entry`` allocated from the mailbox.

Reply:
Your point is well taken. Thank you.